### ISSUE TITLE
Exports should only be in a single location

### ISSUE THREAD

**Issue by EdJoPaTo** [ratatui-org/ratatui#1210]
Currently Terminal is exported both as `ratatui::Terminal` and `ratatui::terminal::Terminal`. (There might be more cases like this, haven't checked yet)

> would be breaking, but the public export should only exist in a single location. Either it's something global or it needs structuring in modules. Allowing for both seems strange. General thought here: reduce the stuff that is publicly accessible. One of the two places should be gone.

_Originally posted by @EdJoPaTo in https://github.com/ratatui-org/ratatui/pull/1180#discussion_r1647468130_

**Comment by orhun** [comment]
What would be the best action for this? I realized we are already hiding the exposed struct via `#[doc(inline)]` which makes it not show up in crate docs. We can probably do the same for other duplicate entries (if any) but I'm not sure if it is a big deal.

**Comment by EdJoPaTo** [comment]
If it's hidden, then it can also be removed. There should be exactly one export, not multiple ones.

**Comment by orhun** [comment]
That would be a breaking change. There are projects which use `ratatui::Terminal` and `ratatui::terminal::Terminal`. Unfortunately it is not consistent.

I propose we make the following types only importable from `ratatui` directly instead of `ratatui::terminal` and document it as a breaking change.

```rust
pub use terminal::{CompletedFrame, Frame, Terminal, TerminalOptions, Viewport};
```

What do you think?

**Comment by EdJoPaTo** [comment]
I'm fine with both on their own, as long as it's a consistent one and not duplicated in two locations. There are arguments for both, so I would be fine with both approaches.

**Comment by joshka** [comment]
It's a pretty common pattern for libraries to expose their main types in several locations. In particular you might see some_crate::error::Error as a good example of this. One thing this does give you is an ability to namespace your type refs easily by importing the module name (as e.g. `terminal::Frame`, `terminal::Terminal`), but perhaps this is fine as `ratatui::Frame` `ratatui::Terminal`?

**Comment by EdJoPaTo** [comment]
> It's a pretty common pattern for libraries to expose their main types in several locations.

Then I don't like their code quality :innocent:

Having exactly one way of doing something improves readability and is more clear. There is no need to question which way to choose from or where the differences might be. And when reading code there is no need to look in different positions for the same thing because people might have used different ways for the same thing.

And in case of functions it prevents fixing bugs in one place but not the other. (like https://github.com/ratatui-org/ratatui/pull/1249#discussion_r1698981102, partially addressed in #1259). But this issue is about the module exports, hopefully soon fixed by #1260 :sparkles:

**Comment by EdJoPaTo** [comment]
- `ratatui::widgets::block::Block` and `ratatui::widgets::Block`
- `ratatui::widgets::block::title::Title` and `ratatui::widgets::block::Title`

**Comment by joshka** [comment]
Yeah, that one is on stalled because I haven't yet gotten back to reading it again.

**Comment by EdJoPaTo** [comment]
> Yeah, that one is on stalled because I haven't yet gotten back to reading it again.

Title but not Block itself?
